tensorizer/_NumpyTensor.py

```diff
@@ -277,21 +277,21 @@ def _decode_torch_dtype(self) -> torch.dtype:
         dtype = _DECODE_MAPPING.get(self.torch_dtype)
         if dtype is not None:
             return dtype
-        else:
-            # Long route using getattr(), any other type
-            if not self.torch_dtype:
-                raise ValueError("Cannot decode an empty dtype.")
-            if not isinstance(self.torch_dtype, str):
-                raise TypeError("torch_dtype must be a string.")
-            module, *dtype_name = self.torch_dtype.split(".", 1)
-
-            # Ensure that it's actually "torch.something"
-            if module != "torch" or len(dtype_name) != 1:
-                raise ValueError(f"Invalid torch_dtype: {self.torch_dtype}")
-
-            dtype = getattr(torch, dtype_name[0])
-            # Ensure that it's a real dtype
-            if dtype is None or not isinstance(dtype, torch.dtype):
-                raise ValueError(f"Invalid torch_dtype: {self.torch_dtype}")
 
-            return dtype
+        # Long route using getattr(), any other type
+        if not self.torch_dtype:
+            raise ValueError("Cannot decode an empty dtype.")
+        if not isinstance(self.torch_dtype, str):
+            raise TypeError("torch_dtype must be a string.")
+        module, *dtype_name = self.torch_dtype.split(".", 1)
+
+        # Ensure that it's actually "torch.something"
+        if module != "torch" or len(dtype_name) != 1:
+            raise ValueError(f"Invalid torch_dtype: {self.torch_dtype}")
+
+        dtype = getattr(torch, dtype_name[0])
+        # Ensure that it's a real dtype
+        if dtype is None or not isinstance(dtype, torch.dtype):
+            raise ValueError(f"Invalid torch_dtype: {self.torch_dtype}")
```

Comment:
Slight bug in my code here. I think this was intended to raise a `ValueError` if `torch.<whatever>` doesn't exist, but instead it throws an `AttributeError` on `getattr` since no default is defined. The `AttributeError` is actually pretty informative, so the below change adds it to the traceback of the intended `ValueError` and makes it distinct from the wrong-type case, which is given a more descriptive error message as well.
```suggestion
        try:
            dtype = getattr(torch, dtype_name[0])
            # Ensure that it's a real dtype
            if not isinstance(dtype, torch.dtype):
                raise TypeError(
                    "Provided torch_dtype is not an instance of torch.dtype"
                    f" (type: {type(dtype).__name__})"
                )
        except (AttributeError, TypeError) as e:
            raise ValueError(f"Invalid torch_dtype: {self.torch_dtype}") from e
```